hooks/build

```diff
@@ -144,6 +144,14 @@ echo
 # So for example 'latest-3.6' will produce PYTHON_VERSION=3.6
 export PYTHON_VERSION=$(echo "${LOCAL_IMAGE}" | rev | cut -d '-' -f1 | rev )
 
+
+# In case of CRON jobs on Travis we run builds without cache
```

Comment:
@fokko -> this will take care about daily running full build from scratch on Travis. I think that might address a lot of your concerns regarding "will it build from scratch". No caches are used for that build, no images are pulled from DockerHub, it will just reinstall everything from the scratch. I guess we can setup notification to send error messages to commit list (or maybe it will happen automatically). This way commiters might be warned about broken transitive dependencies without actually breaking opened pull requests (people will continue to use "working" versions of pip dependencies from last successfully built DockerHub master build. When it gets fixed by committers and merged into master, it will build and new set of dependencies will be used by everyone. I think it will work in self-healing way with minimal disruptions for contributors and committers (and maintainers).

I scheduled daily build at my Travis CI copy and will let you know when it runs so that you can take a look
https://travis-ci.org/potiuk/airflow

Reply:
Here is an example of such Cron job @Fokko: https://travis-ci.org/potiuk/airflow/jobs/509211696. It can run daily on travis and it will run all the tests on a "from the scratch" build. It takes 7:20 minutes to build the image from scratch on Travis worker (per job) but we have clear indication whether clean build of Docker works from the scratch with latest apt-get and pip dependencies and whether all tests are successful. 

This way all the PR will use much faster built incremental image, but we also have full, longer build that is performed automatically and regularly.

At the same time such daily build - since it will be build daily (for example) might be an early indication that one of the transitive dependencies is now failing the installation (but it will not disrupt tests for opened PRs because they will use latest "correctly building" dependencies from the last successful master build.

I think this is a very nice way of addressing both problems - stability for people working on their PRs and assurance for maintainers that the build is "safe". 

I think if we do it this way, we might even not need to solve the transient dependencies case for work in progress - we could focus only on making sure that there is a stable set of dependencies for official releases somehow.